src/loop_logic.rs

```diff
@@ -358,23 +358,58 @@ impl<'l, Data> EventLoop<'l, Data> {
     }
 
     fn dispatch_idles(&mut self, data: &mut Data) {
-        let idles = ::std::mem::take(&mut *self.handle.inner.idles.borrow_mut());
+        // Collect all idle callbacks to prevent RefCell double borrows.
+        let idles = {
+            let mut idles = self.handle.inner.idles.borrow_mut();
+
+            // Idle callbacks are oneshot, replace the Vec
+            mem::take(&mut *idles)
+        };
```

Comment:
Can't we just do this ?

```rust
let idles = mem::take(&mut *self.handle.inner.idles.borrow_mut());
```

Reply:
The intention was to be explicit with ending the scope of the refcell borrows.

I can probably move to using just using drop(cell)

Reply:
IMO the that that we're collecting the callbacks out, with the comment explaining why is already explicit enough. Those redundant scopes are just making the whole code heavier.